lib/comgr/CMakeLists.txt

```diff
@@ -19,24 +19,8 @@ include_directories(${LLVM_INCLUDE_DIRS})
 link_directories(${LLVM_LIBRARY_DIRS})
 include_directories(${CLANG_INCLUDE_DIRS})
 
-# FIXME: There is no way to directly determine the include paths for LLD when
-# working with an LLVM build tree, but we want to avoid having to install LLVM
-# to build Comgr. This should eventually be fixed in usptream LLD so this can
-# be removed.
-if(DEFINED LLVM_BUILD_MAIN_SRC_DIR)
-  set(INTERNAL_LLD_INCLUDE_DIRS "${LLVM_BUILD_MAIN_SRC_DIR}/tools/lld/include")
-  set(EXTERNAL_LLD_INCLUDE_DIRS "${LLVM_BUILD_MAIN_SRC_DIR}/../lld/include")
-  if (EXISTS "${INTERNAL_LLD_INCLUDE_DIRS}"
-      AND IS_DIRECTORY "${INTERNAL_LLD_INCLUDE_DIRS}")
-    set(LLD_INCLUDE_DIRS "${INTERNAL_LLD_INCLUDE_DIRS}")
-  elseif (EXISTS "${EXTERNAL_LLD_INCLUDE_DIRS}"
-      AND IS_DIRECTORY "${EXTERNAL_LLD_INCLUDE_DIRS}")
-    set(LLD_INCLUDE_DIRS "${EXTERNAL_LLD_INCLUDE_DIRS}")
-  else()
-    message(FATAL_ERROR "You are attempting to compile using an LLVM build tree, but the LLD include directory could not be located. The paths '${INTERNAL_LLD_INCLUDE_DIRS}' and '${EXTERNAL_LLD_INCLUDE_DIRS}' were tried.")
-  endif()
-  include_directories(${LLD_INCLUDE_DIRS})
-endif()
+set(LLD_INCLUDE_DIRS "${LLVM_INCLUDE_DIRS}/lld")
```

Comment:
My first concern with this approach is that `LLVM_INCLUDE_DIRS` may be a list of directories. I believe this will result in `"path1;path2/lld"`, and without some check for whether the resulting directory or directories exist it seems fragile.

The current approach is also fragile, but it instead relies on `LLVM_BUILD_MAIN_SRC_DIR` which is defined explicitly for build trees only, and it finds LLD for both the nested build of LLVM and for the new llvm-project monorepo build.

In the long term I think an `LLDConfig.cmake` for the build tree would be the ideal approach, but that needs to be done in upstream LLD and is not a priority.

Reply:
> My first concern with this approach is that `LLVM_INCLUDE_DIRS` may be a list of directories. I believe this will result in `"path1;path2/lld"`, and without some check for whether the resulting directory or directories exist it seems fragile.

`FOREACH` can be used to address that concern; I've updated the PR accordingly.
```